client/disputes/evidence/index.js

```diff
@@ -353,252 +47,85 @@ const getDisputeProductType = ( dispute ) => {
 
 // Temporary MVP data wrapper
 export default ( { query } ) => {
-	const path = `/wc/v3/payments/disputes/${ query.id }`;
+	const { id: disputeId } = query;
+	const { dispute, isLoading, updateDispute } = useDispute( disputeId );
 
-	const [ dispute, setDispute ] = useState();
-	const [ loading, setLoading ] = useState( false );
-	const [ evidence, setEvidence ] = useState( {} ); // Evidence to update.
 	const {
-		createSuccessNotice,
-		createErrorNotice,
-		createInfoNotice,
-	} = useDispatch( 'core/notices' );
-
-	const pristine =
-		! dispute ||
-		isMatchWith(
-			dispute.evidence,
-			evidence,
-			( disputeValue, formValue ) => {
-				// Treat null and '' as equal values.
-				if ( null === disputeValue && ! formValue ) {
-					return true;
-				}
-			}
-		);
-
-	const confirmationNavigationCallback = useConfirmNavigation( () => {
-		if ( pristine ) {
-			return;
-		}
-
-		return __(
-			'There are unsaved changes on this page. Are you sure you want to leave and discard the unsaved changes?',
-			'woocommerce-payments'
-		);
-	} );
-
-	useEffect( confirmationNavigationCallback, [
-		pristine,
-		confirmationNavigationCallback,
-	] );
+		isSavingEvidence,
+		isUploadingEvidence,
+		evidenceTransient,
+		evidenceUploadErrors,
+		saveEvidence,
+		submitEvidence,
+		updateEvidenceTransientForDispute,
+		updateEvidenceUploadErrorsForDispute,
+		uploadFileEvidenceForDispute,
+	} = useDisputeEvidence( disputeId );
+
+	const { setMessage: setNavigationMessage } = useConfirmNavigation();
```

Comment:
You'll notice that the `useConfirmNavigation` hook has been changed significantly, most importantly in a way such that the navigation message state is managed in the hook. You can see that more clearly in the corresponding file.